src/touch.js

```diff
@@ -0,0 +1,89 @@
+var common = require('./common');
+var fs = require('fs');
+
+module.exports = _touch;
+
+//@
+//@ ### touch(file)
+//@
+//@ Examples:
+//@
+//@ ```javascript
+//@ touch('file.txt');
+//@ ```
+//@
+//@ Creates a file if it does not exist. Sets the file's mtime and ctime to NOW if the file
+//@ does exist
+function _touch(opts, file) {
+	opts = common.parseOptions(opts, {
+		'a': 'atime',
+		'c': 'no_create',
+		'd': 'date',
+		'm': 'mtime',
+		'r': 'reference',
+	});
+
+	if (!file) {
+		common.error('no paths given');
+	}
+
+	if (typeof file !== 'string') {
+		common.error('file arg should be a single file path');
+	}
+
+	// if the string ends with "/" then stop
+	if (file.indexOf('/', file.length - 1) !== -1) {
+		common.error('cannot touch directory ' + file);
+	}
+
+	if (fileExists(file) && fs.statSync(file).isDirectory()) {
+		common.error('cannot touch directory ' + file);
+	}
+
+	// if the file doesn't already exist and the user has specified --no-create then
+	// this script is finished
+	if (!fileExists(file) && opts.no_create) {
+		return;
+	}
+
+	// open the file and then close it. this will create it if it doesn't exist but will
+	// not wipe out it's contents
+	fs.closeSync(fs.openSync(file, 'a+'));
```

Comment:
I'd go for the actual flags used by `touch` from coreutils: `O_WRONLY | O_CREAT` (they are available in `require('constants')`).